cmds/pxeboot/pxeboot.go

```diff
@@ -69,13 +76,13 @@ func Netboot() error {
 		}
 
 		log.Printf("Attempting to get DHCP lease on %s", iface.Attrs().Name)
-		packet := attemptDHCPLease(iface, 30*time.Second, 5)
-		if packet == nil {
-			log.Printf("No lease on %s", iface.Attrs().Name)
+		packet, err := attemptDHCPLease(iface, 30*time.Second, 5)
+		if packet == nil || err != nil {
```

Comment:
nit (i.e. feel free to ignore or address in another PR): I normally prefer that only `err` has to be checked, and if for some reason `packet` is nil, it bombs out and we're sure it's a bug in the underlying call.

In this case it could fail in two points inside `attemptDHCPLease`: either the return value from `client.Request` or from `dhclient.NewPacket`. The non-`nil`ness could be enforced either here or inside the called functions; and a proper error could be returned (e.g. `Packet is nil`). So the caller can just check for `err` and not having to infer what the error is when `err` is `nil` and `packet` is `nil` too

Reply:
actually, there's no space for nits in this PR, let's get things done first and clean them up later or this will grow bigger and bigger.

Reply:
it's already an or<packet, error>, so i just removed the nil packet check.

side note: this is why i want Go to have a nicer type system. algebraic types...